ports/stm32/mboot/Makefile

```diff
@@ -6,7 +6,7 @@ BOARD ?= $(notdir $(BOARD_DIR:/=))
 else
 # If not given on the command line, then default to PYBV10.
 BOARD ?= PYBV10
-BOARD_DIR ?= $(abspath ../boards/$(BOARD))
+BOARD_DIR ?= ../boards/$(BOARD)
```

Comment:
@Iksas now that this PR is merged and I've been using it, I see that this change here is no good: the build artefacts (object files) for boards with source files are placed outside the build directory.  That's because this variable starts with `..`.

For the list of source files added to `SRC_C`, none of them can start with `..`.  The usual fix for that would be to make the files relative to the top of the MicroPython repo (because of the vpath), eg `ports/stm32/boards/$(BOARD)`.  But then the `$(wildcard ...)` pattern won't find files in this directory.

I'm not sure how to fix this.  Did you have an alternative way to make this work?

Reply:
I'll try to investigate this problem on my machine.

Reply:
This scenario causes multiple conflicts between the `subst` and `wildcard` commands. To solve this, it probably will be necessary to perform some string manipulations with `sed`.

Implementing those changes, especially while keeping the file readable, will not be straight-forward. I'd recommend to remove all changes from this file until a tidy solution can be found in the future.

Reply:
I have reverted this line in #13511.